### PR TITLE
Clarify `cond` to `If` must contain a single element

### PR DESCRIPTION
**Pull request by justinchuby**
### Description

Clarify `cond` to `If` must contain a single element. Previously it is unclear what the condition can be.

Fix reference runtime accordingly.

### Motivation and Context

Fixes https://github.com/onnx/onnx/issues/5520

### CODE REVIEW

File: onnx/reference/ops/op_if.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 # SPDX-License-Identifier: Apache-2.0
 
+import numpy as np
```
Comment (justinchuby) [inline onnx/reference/ops/op_if.py:5]:
```suggestion
from __future__ import annotations

import numpy as np
```

File: onnx/reference/ops/op_if.py [hunk 2]
```diff
@@ -23,28 +26,20 @@ def need_context(self) -> bool:
...
+    def _run(
+        self,
+        cond: np.ndarray | np.bool_,
+        context=None,
+        else_branch=None,
+        then_branch=None,
+        attributes=None,
+    ):
+        if cond.size != 1:
+            raise RuntimeError(
```
Comment (justinchuby) [inline onnx/reference/ops/op_if.py:38]:
```suggestion
            raise ValueError(
```

File: onnx/reference/ops/op_if.py [hunk 3]
```diff
@@ -23,28 +26,20 @@ def need_context(self) -> bool:
...
+        self,
+        cond: np.ndarray | np.bool_,
+        context=None,
+        else_branch=None,
+        then_branch=None,
+        attributes=None,
+    ):
+        if cond.size != 1:
+            raise ValueError(
+                f"Operator If ({self.onnx_node.name!r}) expects a single element condition, but size of 'cond' is {len(cond)}."
```
Comment (justinchuby) [inline onnx/reference/ops/op_if.py:39]:
```suggestion
                f"Operator If ({self.onnx_node.name!r}) expects a single element as condition, but the size of 'cond' is {len(cond)}."
```

**Review by jcwchen — COMMENTED** [review]
Is it worthy to add such a check (required single element for cond) in shape inference? Not sure whether it will break any existing usage tho.

**Comment by justinchuby** [comment]
> Is it worthy to add such a check (required single element for cond) in shape inference? Not sure whether it will break any existing usage tho.

Since the shape may be unknown I am not sure if the check is robust. I can do this as a follow up. Thanks!

**Pull request merged**